src/Services/AppointmentService.php

```diff
@@ -18,14 +18,19 @@
 use OpenEMR\Common\Database\QueryUtils;
 use OpenEMR\Common\Uuid\UuidRegistry;
 use OpenEMR\Services\Search\DateSearchField;
+use OpenEMR\Services\Search\FhirSearchWhereClauseBuilder;
 use OpenEMR\Services\Search\TokenSearchField;
 use OpenEMR\Services\Search\TokenSearchValue;
+use OpenEMR\Validators\ProcessingResult;
 use Particle\Validator\Exception\InvalidValueException;
 use Particle\Validator\Validator;
 
 class AppointmentService extends BaseService
```

Comment:
The functions that collect the appointment will break on recurrent appointments (this is because a recurrent appointment takes up only 1 row in the table). There is a very complicated function call that deals with this (in past there were many places where direct calls to the table were breaking on recurrent appointments which were then consolidated to use this function) at:
https://github.com/openemr/openemr/blob/master/library/appointments.inc.php#L366

Reply:
@bradymiller This sounds like an existing problem with the AppointmentService in general right?  The appointment apis were built before I came on the scene so I'm just trying to figure out if this is an existing bug or a new one I introduced.

Reply:
yep, this is an existing bug that will impact any call to the functions in this service class that collect appointments. At this point looks like just impacting calls to api. Would just post this as a issue and not fix in this PR since the fix may require some thought anyways especially on the euuid front.